clients/vault/src/oracle/collector/proof_builder.rs

```diff
@@ -42,17 +43,36 @@ pub enum ProofStatus {
 	NoTxSetFound(Slot),
 }
 
+const MAX_SLOT_TO_REMEMBER : u64 = 12;
 // handles the creation of proofs.
 // this means it will access the maps and potentially the files.
 impl ScpMessageCollector {
 	fn get_slot(&self, tx_hash: &TxHash) -> Option<Slot> {
 		self.tx_hash_map().get(tx_hash).map(|slot| *slot)
 	}
 
+	
 	/// Returns either a list of ScpEnvelopes or a ProofStatus saying it failed to retrieve a list.
 	fn get_envelopes(&self, slot: Slot) -> Result<UnlimitedVarArray<ScpEnvelope>, ProofStatus> {
-		let (envelopes, is_from_file) =
-			self._get_envelopes(slot).ok_or(ProofStatus::NoEnvelopesFound(slot))?;
+		let envelopes =
+			self._get_envelopes(slot);
+		
+		if let None = envelopes{
+
+			let last_slot_index = *self.last_slot_index();
+			let action_sender = self.action_sender.clone();
+			if last_slot_index - MAX_SLOT_TO_REMEMBER < slot {
+				tokio::spawn(async move { 
+					action_sender
+					.send(ActorMessage::GetScpState { missed_slot : slot })
+					.await
+				});
+			}
+
+			return Err(ProofStatus::NoEnvelopesFound(slot))
```

Comment:
Maybe change this to return `ProofStatus::LackingEnvelopes` and remove the `ProofStatus::NoEnvelopesFound(slot)` completely. I'd argue that having both variants in the enum is not really beneficial.

Reply:
Done.